iree/compiler/Dialect/VM/Conversion/VMToEmitC/ConvertVMToEmitC.cpp

```diff
@@ -28,6 +28,37 @@ namespace iree_compiler {
 
 namespace {
 
+LogicalResult sanitizeSymbolNames(IREE::VM::ModuleOp &moduleOp) {
+  auto replaceIllegalChars = [](std::string &symbolName) {
+    // replace illegal characters in symbol name with "_"
+    std::replace(symbolName.begin(), symbolName.end(), '>', '_');
+    std::replace(symbolName.begin(), symbolName.end(), '<', '_');
+  };
+
+  std::string symbolName(moduleOp.getName());
+
+  // Sanitize name
+  replaceIllegalChars(symbolName);
+
+  if (failed(moduleOp.replaceAllSymbolUses(
+          StringAttr::get(moduleOp.getContext(), symbolName), moduleOp)))
+    return failure();
```

Comment:
```suggestion
```

The module name shouldn't be used in the VM IR anyway, so it should be safe to just rename it. (In the C Target it gets used as a prefix for function names, global variables, struct definitions etc.)

Reply:
The module name must make it unmodified to `iree_vm_module_name` - if you are changing that here then you'll need to ensure it makes its way into the right struct at https://github.com/google/iree/blob/53d5090cefae9316754152ab557a392da39a1e16/iree/compiler/Dialect/VM/Target/C/CModuleTarget.cpp#L239. This is the user-visible name that is used to lookup and link across modules.